src/main/io/osd.c

```diff
@@ -460,6 +462,37 @@ static bool osdDrawSingleElement(uint8_t item)
     char buff[OSD_ELEMENT_BUFFER_LENGTH] = "";
 
     switch (item) {
+    case OSD_ARROW_FLOAC: 
+        {
+            const int angleR = attitude.values.roll;
+            const int angleP = attitude.values.pitch; // still gotta update all angleR and angleP pointers.
+            if (IS_RC_MODE_ACTIVE(BOXFLIPOVERAFTERCRASH)) {
+                if (angleP > 0 && ((angleR > 175 && angleR < 180) || (angleR > -180 && angleR < -175))) {
+			        buff[0] = SYM_ARROW_SOUTH;
+			    }
+                else if (angleP > 0 && angleR > 0 && angleR < 175) {
+                    buff[0] = (SYM_ARROW_EAST + 2);
+                }
+                else if (angleP > 0 && angleR < 0 && angleR > -175) {
+                    buff[0] = (SYM_ARROW_WEST + 2);
+                }
+                else if (angleP <= 0 && ((angleR > 175 && angleR < 180) || (angleR > -180 && angleR < -175))) {
+                    buff[0] = SYM_ARROW_NORTH;
+                }
+                else if (angleP <= 0 && angleR > 0 && angleR < 175) {
+                    buff[0] = (SYM_ARROW_NORTH + 2);
+                }
+                else if (angleP <= 0 && angleR < 0 && angleR > -175) {
+                    buff[0] = (SYM_ARROW_SOUTH + 2);
+                }
+            }
+            else
+            {
+                buff[0] = ' ';
+                buff[1] = '\0';
```

Comment:
This needs to go to before the `break` - the string needs to be terminated in the case when an arrow is printed as well.

Reply:
How do you mean, before the break? How I see it it comes before break. 
The string termination of later arrow and name change will be done as well once I fix all requested changed.

Reply:
What I mean is that it has to be moved down to just before the `break` statement, so that it's applied irrespective of which branch is taken.